### ISSUE TITLE
Ballot form should be customizable

### ISSUE THREAD

**Issue by tienne-B** [TabbycatDebate/tabbycat#955]
Similarly to feedback forms, ballot forms should be customizable with extra "questions." This task is to merge #378 and #771, both of them proposing a widget (number field / checkbox) for the ballot form. Are those comparable tasks?

**Comment by tienne-B** [comment]
Don't think it is a problem, as these questions are in supplement of the information requested by the debate format, and so I can't see incompatibilities. For score range, that could be done with simple form validation.

This approach is not based on requirements of a debate format, rather, to link two similar ideas (both with examples showing need for flexibility) under the same implementation to reduce complexity.

**Comment by philipbelesky** [comment]
So, one concern would be that configuration of questions in this manner is relatively advanced use case, particularly in the case of ballots where (unlike feedback) there is an assumption that they will 'just work' according to the format's settings. 

Additionally, even if this method was available I'm struggling to think of use cases that wouldn't be better served by adapting the form automatically based on the provided settings. For example in #378 it makes sense to automatically offer the choice to record that data and people can choose not to enter it if they don't care. #771 is a case where it would benefit from customisability, but that use case seems sufficiently complex to preclude a more generic solution like this issue seems to suggest — i.e. the `ScoreCriterion` model — partricularly when considering how the information is displayed rather than just recorded.

CZ probably has more thoughts on the feasability of a technical implementation, although I'll note that the new version of the ballot form is less directly tied to django's forms which could be challenging. From a more UX perspective I'm a little unclear on what benefit this provides that makes it worth the effort. Are there other kinds of formats where this kind of information needs to be collected in this way?

**Comment by czlee** [comment]
Yeah, scoresheets are aggregated using specific rules about how debate formats work, which we can abstract (and have abstracted) to a degree, but would be extremely complex to implement more flexibly, both to program and in terms of user setup. Unless we want to take information and not do anything with it, but I assume at a minimum we want the user to be able to set up validation checks. In the case of #771, for example, we add components into a total speaker score (I think?). Absent a few more debate styles that would fit into this framework, I sort of think this would add more complexity than it would save.
